src/main/java/io/aiven/kafka/connect/common/grouper/TopicPartitionRecordGrouper.java

```diff
@@ -66,20 +66,21 @@
         Objects.requireNonNull(filenameTemplate, "filenameTemplate cannot be null");
         Objects.requireNonNull(tsSource, "tsSource cannot be null");
         this.filenameTemplate = filenameTemplate;
-        this.setTimestampBasedOnRecord = record -> new Function<>() {
-            private final Map<String, DateTimeFormatter> timestampFormatters =
+
+        final Map<String, DateTimeFormatter> timestampFormatters =
```

Comment:
This `Map` could `private static final`. I don't think each instance of `TopicPartitionRecordGrouper` needs its own `Map`. `DateTimeFormatter` is thread-safe, so no problem caused by concurrency.

Reply:
And the values of `Map` can be constants also to reuse them in tests where they a copy-pasted.

Reply:
Made `private static final`